core/runtime/wavm/module.cpp

```diff
@@ -37,6 +39,16 @@ namespace kagome::runtime::wavm {
       return nullptr;
     }
 
+    auto imports = WAVM::Runtime::getModuleIR(module).memories.imports;
+    if (imports.empty()) {
+      logger->critical("Memory type not found in imports.");
+      return nullptr;
+    }
+
+    IntrinsicModule::kIntrinsicMemoryType = imports[0].type;
```

Comment:
Oof, I don't like this solution, mutable global state is evil. Could you at least refactor it out to some separate class which would store it? That would be more explicit and controllable, I believe.

Reply:
Doesn't